pkg/middleware/utils.go

```diff
@@ -86,3 +87,27 @@ func GetExperiment(r *http.Request) (*config.OptimizelyExperiment, error) {
 	}
 	return experiment, nil
 }
+
+// CoerceType coerces typed value from string
+func CoerceType(s string) interface{} {
+
+	if u, err := strconv.Unquote(s); err == nil {
+		if u != s {
+			return u
+		}
+	}
+
+	if i, err := strconv.ParseInt(s, 10, 0); err == nil {
+		return i
+	}
+
+	if d, err := strconv.ParseFloat(s, 64); err == nil {
+		return d
+	}
+
+	if b, err := strconv.ParseBool(s); err == nil {
```

Comment:
nit: knowing that ParseBool accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False, I think maybe we should limit inputs to true/false only , otherwise looks good